lib/rules.js

```diff
@@ -184,6 +185,234 @@ export const MULTIPLY_BY_INVERSE = defineRuleString('#a / (#b / #c)', '#a * (#c
+
+        if (query.getValue(numerator) > query.getValue(denominator)
+            &&  query.getValue(denominator) == 1){
+            result = build.applyNode('pow', [variable, numerator])
+        } else if (query.getValue(numerator) === query.getValue(denominator)){
+            result = build.applyNode('pow', [variable, build.numberNode(1)])
+        } else {
+            result = build.applyNode (
+                {type: 'Identifier', name: 'nthRoot'},
+                [build.applyNode('pow', [variable, numerator]), denominator]
+            )
+        }
+        return result
+    }
+)
+
+// e.g. nthRoot(2, 2) * nthRoot(3, 2) -> nthRoot(2 * 3, 2)
+export const COMBINE_UNDER_ROOT = defineRuleString('nthRoot(#a, #c) * nthRoot(#b, #c)', 'nthRoot(#a * #b, #c)')
+
+// e.g. 2 * 2 * 2 -> 2 ^ 3
+export const CONVERT_MULTIPLICATION_TO_EXPONENT = defineRuleString('#a^#b_0 * ...', '#a^#eval(#b_0 + ...)', {a: query.isNumber, b: query.isNumber})
+
+// e.g. nthRoot(2 * x) -> nthRoot(2) * nthRoot(x)
+export const DISTRIBUTE_NTH_ROOT = defineRule(
+    (node) => {
+        let isNthRoot = false
+        let validRadicand = false
+        if (query.isApply(node)){
+            isNthRoot = query.isIdentifier(node.op)
+                && node.op.name == 'nthRoot'
+            validRadicand = query.isMul(node.args[0])
+        }
+        return (isNthRoot && validRadicand) ? {node} : null
+    },
+
+    (node) => {
+        const radicand = node.args[0]
+        const result = build.applyNode(
+            'mul',
+            radicand.args.map(arg => {
+                return build.applyNode(
+                    {type: 'Identifier', name: 'nthRoot'},
+                    [arg]
+                )
+            })
+        )
+        return result
+    }
+)
+
+// e.g. nthRoot(4) * nthRoot(x^2) -> 2 * x
+export const EVALUATE_DISTRIBUTED_NTH_ROOT = defineRule(
+    (node) => {
+        let isDistributed = query.isMul(node)
+        let canEvaluate = canApplyRule(CANCEL_EXPONENT, node)
+        return (isDistributed && canEvaluate) ? {node} : null
+    },
+
+    (node) => {
+        const result = build.applyNode(
+            'mul',
+            node.args.map(nthRoot => {
+                if (query.isNumber(nthRoot.args[0])) {
+                    return applyRule(NTH_ROOT_VALUE,
+                                     build.applyNode(
+                                         {type: 'Identifier', name: 'nthRoot'},
+                                         [nthRoot.args[0]]
+                                     ))
+                }
+                return applyRule(CANCEL_EXPONENT,
+                                 build.applyNode(
+                                     {type: 'Identifier', name: 'nthRoot'},
+                                     [nthRoot.args[0]]
+                                 ))
+            })
+        )
+        return result
+    }
+)
+
+export const primeFactorization = (number, result = []) => {
```

Comment:
Would be nice to reuse https://github.com/semantic-math/math-evaluator/blob/master/lib/functions.js#L1-L20.